move SymInt and SymIntArrayRef to c10/core

Krovatkin opened the pull request:
Fixes #ISSUE_NUMBER

Krovatkin commented:
@malfet I finally got around to move SymInt back to c10/core. Could you please advise if there are any lists any of these files need to be added and the old ones removed off? Thank you!

malfet reviewed (approved):
LGTM, but please import this change to phab and make sure builds are passing

Krovatkin committed:
move SymInt and SymIntArrayRef to c10/core (#77009)

Summary:
Fixes #ISSUE_NUMBER

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77009

Reviewed By: malfet

Differential Revision: D36267441

Pulled By: Krovatkin

fbshipit-source-id: 7b94ca835958a90f03e63dc659fa9a1e478ce17b
